### ISSUE TITLE
avoid using redundant code (train_pn & train_rpn)

### ISSUE THREAD

**Issue by Abdoul1710** [Cortys/SEP-Probly-WS2526#518]
in probly.train.evidential.torch there are the two functions train_pn and train_rpn that are used to compute the loss for pn and rpn (another pipeline than the other modes since they need ood). try merging these three functions into 1 by creating loss functions out of train_pn & train_rpn
